test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -86,11 +87,15 @@ private void assertCommandBehavior(String inputCommand,
         if(isRelevantPersonsExpected){
             assertEquals(lastShownList, r.getRelevantPersons().get());
         }
+        
+        Command command = new Parser().parseCommand(inputCommand);
 
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if (command.isMutating()){
+            assertEquals(addressBook,saveFile.load());
```

Comment:
it would also be good to check if the command that is NOT mutating indeed doesn't save the file